plutus-conformance/src/PlutusConformance/Common.hs

```diff
@@ -69,62 +71,47 @@ shownEvaluationFailure = "evaluation failure"
 
 -- For UPLC evaluation tests
 
-type UplcProg = UPLC.Program Name DefaultUni DefaultFun ()
-
-termToProg :: UPLC.Term Name DefaultUni DefaultFun () -> UplcProg
-termToProg = UPLC.Program () (defaultVersion ())
-
--- | Our `runner` for the UPLC tests is the CEK machine.
-evalUplcProg :: UplcProg -> Maybe UplcProg
-evalUplcProg p =
-    let eitherExceptionProg =
-            fmap
-                termToProg
-                (evaluateCekNoEmit defaultCekParameters (UPLC._progTerm p))
-    in
-        case eitherExceptionProg of
-            Left _     -> Nothing
-            Right prog -> Just prog
-
-{- | Run an input with the `runner` and return the result, in `Text`.
-When fail, the result is either the default text for parse error or evaluation failure. -}
-mkResult ::
-    (UplcProg -> Maybe UplcProg) -- ^ The `runner` to run the test inputs with.
-    -> Either ParserErrorBundle (UPLC.Program Name DefaultUni DefaultFun SourcePos)
-    -- ^ The result of parsing.
-    -> IO T.Text -- ^ The result of the `runner` in `Text`.
-mkResult _ (Left (ParseErrorB _err)) = pure shownParseError
-mkResult runner (Right prog)        = do
-    -- catch exceptions from `runner` and keep going unless it's an async exception.
-    maybeException <- try (evaluate $ force $ runner (() <$ prog)):: IO (Either SomeException (Maybe UplcProg))
-    case maybeException of
-        Left _         -> pure shownEvaluationFailure
-        -- it doesn't matter how the evaluation fail, they're all "evaluation failure"
-        Right Nothing  -> pure shownEvaluationFailure
-        Right (Just a) -> pure $ display a
-
 -- | The default parser to parse the inputs.
 parseTxt ::
     T.Text
     -> Either ParserErrorBundle (UPLC.Program Name DefaultUni DefaultFun SourcePos)
 parseTxt resTxt = runQuoteT $ UPLC.parseProgram resTxt
 
--- | Build a test given the runner and a `TestContent`.
--- TODO maybe abstract this for other tests too if it takes in `mkResult`.
-mkUplcTestCase :: (UplcProg -> Maybe UplcProg) -> TestContent -> TestTree
-mkUplcTestCase runner test =
-    testCase (testName test) (
-        do r <- mkResult runner (parseTxt (inputProg test))
-           expected test @=? r)
+-- | Our `evaluator` for the UPLC tests is the CEK machine.
+evalUplcProg :: UplcEvaluator
+evalUplcProg = traverseOf UPLC.progTerm eval
+  where
+    eval t = do
+        -- The evaluator throws if the term has free variables
+        case UPLC.deBruijnTerm t of
```

Comment:
With this I think I won't need to catch exceptions in the executable either! I'll push the change and check......